--- a/tools/clang/scripts/build.py
+++ b/tools/clang/scripts/build.py
@@ -489,11 +489,7 @@ def main():
   if not args.gcc_toolchain:
     return
   # Find libstdc++.so.6
-  libstdcpp = subprocess.check_output([
-      os.path.join(args.gcc_toolchain, 'bin', 'g++'),
-      '-print-file-name=libstdc++.so.6'
-  ],
-                                      universal_newlines=True).rstrip()
+  libstdcpp = '/build/chromium/parts/chromium/build/third_party/llvm-build-tools/gcc-10.2.0-bionic/lib64/libstdc++.so.6'
 
   EnsureDirExists(os.path.join(build_dir, 'lib'))
   CopyFile(libstdcpp, os.path.join(build_dir, 'lib'))
