Description: Enable VA-API flag on ozone wayland
Author: Jianhui J Dai <jianhui.j.dai@intel.com>
Forwarded: https://chromium-review.googlesource.com/c/chromium/src/+/3584922

diff --git a/ui/ozone/platform/wayland/ozone_platform_wayland.cc b/ui/ozone/platform/wayland/ozone_platform_wayland.cc
index 661859754..42e927b7 100644
--- a/ui/ozone/platform/wayland/ozone_platform_wayland.cc
+++ b/ui/ozone/platform/wayland/ozone_platform_wayland.cc

@@ -289,6 +289,8 @@
       // arbitrary position.
       properties->supports_global_screen_coordinates = false;
 
+      properties->supports_vaapi = true;
+
       initialised = true;
     }
