--- a/components/os_crypt/BUILD.gn
+++ b/components/os_crypt/BUILD.gn
@@ -30,6 +30,9 @@
     libs = [ "gnome-keyring" ]
   }
 }
+pkg_config("libsecret-1") {
+  packages = [ "libsecret-1" ]
+}

 component("os_crypt") {
   sources = [
@@ -101,6 +101,6 @@
         "libsecret_util_linux.h",
       ]
       configs += [ "//build/config/linux:glib" ]
+      configs += [ ":libsecret-1" ]
-      deps += [ "//third_party/libsecret" ]
       defines += [ "USE_LIBSECRET" ]
     }
     if (use_dbus) {
@@ -181,7 +181,7 @@
     }
     if (use_glib) {
       sources += [ "key_storage_libsecret_unittest.cc" ]
+      configs += [ ":libsecret-1" ]
-      deps += [ "//third_party/libsecret" ]
       defines += [ "USE_LIBSECRET" ]
     }
     if (use_dbus) {
