--- a/components/os_crypt/BUILD.gn
+++ b/components/os_crypt/BUILD.gn
@@ -101,7 +101,6 @@
         "libsecret_util_linux.h",
       ]
       configs += [ "//build/config/linux:glib" ]
-      deps += [ "//third_party/libsecret" ]
       defines += [ "USE_LIBSECRET" ]
     }
     if (use_dbus) {
@@ -181,7 +181,6 @@
     }
     if (use_glib) {
       sources += [ "key_storage_libsecret_unittest.cc" ]
-      deps += [ "//third_party/libsecret" ]
       defines += [ "USE_LIBSECRET" ]
     }
     if (use_dbus) {
